### DISCUSSION TITLE
Tryinna style Phlex::Table components. Anyone got a way that works?

### DISCUSSION THREAD

**Discussion by dhnaranjo** [yippee-fun/phlex#305] [Component architecture]
I'm using `Phlex::Table` and it's getting the job done pretty well right up until I want to get styling. I'm Tailwind'n so I gotta hit up individual elements. Here's how I'm trying to accomplish that:

```
class SuperGreatTable < Phlex::View
  include Phlex::Table

  property "I want this text right-aligned" do |content|
    span { content.to_s }
  end

  # This works
  def table
    super(class: "rounded-xl")
  end
  
  # This does not work, tho it sure looks similar in concept
  # def head
  #   super(class: "text-right")
  # end
  
  # Or this, which head aliases to
  # def thead
  #   super(class: "text-right")
  # end

  # This does work.
  def head(&)
    thead(class: "text-right", &)
  end
end
```

It's just kinda not obvious why I can use `super` on table but not on head or thead.

Edit: I landed at where I needed to be like this:

```
class Table < ApplicationView
  include Phlex::Table

  # ...

  def table
    super(class: "w-full rounded-xl border text-xs border-lightGray")
  end

  def head(&)
    thead(class: "bg-opacity-40 text-left bg-twilightPearl", &)
  end

  def head_header(**args, &) = th(class: "p-4 border border-lightGray font-semibold", **args, &)

  def body_cell(&) = td(class: "p-4 border border-lightGray", &)

  # ...
end
```

Which is not like... awful, but it is a bit non-obvious.

**Comment by Bumppoman** [comment]
I'm a little surprised your table method would work like that but in general, the convention that Phlex has established is passing blocks.  I would recommend you change your table to work the same way and save yourself some potential future frustration.

**Comment by joeldrapper** [comment]
Hey @dhnaranjo, `Phlex::Table` is still very experimental and subject to change and I think you've found an important issue. Because of the way it is currently implemented with aliases, it's quite brittle with regards to which methods you override. I think the only way around this is to fix it. I’ll try to get something out in `0.5` though it will still be experimental.

**Reply by dhnaranjo** [reply]
Aw yeah I'm prepared to pay the cost of using prerelease tools.

I think that what left the worst taste in my mouth here was needing to understand the need to pass the splat of arguments on `head_header`, and only because I didn't need to do that elsewhere. I could consistently just forward `(*a, **b, &c)` on everything so that the choice is unambiguous and doesn't like, require research, but that's a lil ugly.
